ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilderTest.java

```diff
@@ -0,0 +1,315 @@
+    when(leftKTableMultiKey.join(any(KTable.class), any(KsqlKeyExtractor.class), any())).thenReturn(resultKTable);
+
+    planBuilder = new KSPlanBuilder(
+        mock(RuntimeBuildContext.class),
+        mock(SqlPredicateFactory.class),
+        mock(AggregateParamsFactory.class),
+        mock(StreamsFactories.class)
+    );
+  }
+
+  @Test
+  public void shouldDoLeftJoinOnNonKey() {
+    // Given:
+    givenLeftJoin(left, JOIN_COLUMN);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTable).leftJoin(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(1)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoLeftJoinOnKey() {
+    // Given:
+    givenLeftJoin(left, L_KEY);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTable).leftJoin(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(2)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoLeftJoinOnSubKey() {
+    // Given:
+    givenLeftJoin(leftMultiKey, L_KEY_2);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTableMultiKey).leftJoin(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(3)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA_MULTI_KEY.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoInnerJoinOnNonKey() {
+    // Given:
+    givenInnerJoin(left, JOIN_COLUMN);
+
+    // When:
+    final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
+
+    // Then:
+    verify(leftKTable).join(
+        same(rightKTable),
+        eq(new KsqlKeyExtractor<>(1)),
+        eq(new KsqlValueJoiner(LEFT_SCHEMA.value().size(), RIGHT_SCHEMA.value().size(), 0))
+    );
+    verifyNoMoreInteractions(leftKTable, rightKTable, resultKTable);
+    assertThat(result.getTable(), is(resultKTable));
+    assertThat(result.getExecutionKeyFactory(), is(executionKeyFactory));
+  }
+
+  @Test
+  public void shouldDoInnerJoinOnKey() {
```

Comment:
Same question as above: What is the purpose of this test? Isn't this a primary key join, not a foreign key join?